libraries/core_libs/consensus/src/transaction_manager/transaction_manager.cpp

```diff
@@ -229,6 +230,27 @@ size_t TransactionManager::getNonfinalizedTrxSize() const {
   return nonfinalized_transactions_in_dag_.size();
 }
 
+std::vector<std::shared_ptr<Transaction>> TransactionManager::getNonfinalizedTrx(const std::vector<trx_hash_t> &hashes,
+                                                                                 bool sorted) {
+  std::vector<std::shared_ptr<Transaction>> ret;
+  ret.reserve(hashes.size());
+  std::shared_lock transactions_lock(transactions_mutex_);
+  for (const auto &hash : hashes) {
+    assert(nonfinalized_transactions_in_dag_.contains(hash));
```

Comment:
this assert is suspicious, what if you dont have the transaction (we used to have missing tx)... You never know what happens on network so if you received pbft block, and you dont have all of the transactions - this will kill the node ?

Reply:
If you do not have trxs, then dag block and it's transactions can not be insert. If this assert is hit, either I have missed some flow or we have a bug. We could change it to error, but I am not checking logs on devnet, so this way I will now sooner. So, if you want I can create task to change it to error log when we want to deploy it on testnet

Reply:
so did you go through the flow and make sure it is not possible or not ?

Reply:
ok make sure it does not get to testnet(I am sure we forget about it so take pls as your responsibility to monitor this) before we test it properly on devnet then.